src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs

```diff
@@ -904,7 +906,14 @@ public string Subject
         /// <returns>Encoded token string without signature or authentication tag.</returns>
         public override string ToString()
         {
-            return EncodedToken.Substring(0, EncodedToken.LastIndexOf("."));
+            if (_lastDot < 0)
+            {
+                _lastDot = EncodedToken.LastIndexOf('.');
```

Comment:
as if there is no dot, _lastDot will be -1.

Reply:
fixed